### ISSUE TITLE
如何通过日志看两方加法发送的数据

### ISSUE THREAD

**Issue by daydayuphere** [secretflow/spu#458]
### Issue Type

Build/Install

### Modules Involved

MPC protocol

### Have you reproduced the bug with SPU HEAD?

Yes

### Have you searched existing issues?

Yes

### SPU Version

spu0.6

### OS Platform and Distribution

centos7

### Python Version

3.8.13

### Current Behavior?
基于spu的nodectl的分布式架构，写了一个两方加法的demo
如何通过日志查看秘密分享发送的数据分片，现在的日志看不出来
[2024-01-03 17:55:19,065] [MainProcess] RunR: builtin_fetch_meta at node:0
[2024-01-03 17:55:19,066] [MainProcess] RunR: builtin_fetch_object at node:0
[2024-01-03 17:55:19,070] [MainProcess] RunR: builtin_fetch_object at node:0
[2024-01-03 17:55:24,987] [MainProcess] Run : builtin_spu_init at node:0
[2024-01-03 17:55:24,987] [MainProcess] spu-runtime (SPU) already exist, reuse it
[2024-01-03 17:55:24,990] [MainProcess] Run : make_rand at node:0
[2024-01-03 17:55:24,995] [MainProcess] Run : make_shares at node:0
[2024-01-03 17:55:24,996] [MainProcess] RunR: builtin_fetch_meta at node:0
[2024-01-03 17:55:25,012] [MainProcess] Run : builtin_spu_run at node:0
[2024-01-03 17:55:25,014] [MainProcess] RunR: builtin_fetch_object at node:0
[2024-01-03 17:55:25.015] [info] [api.cc:158] [Profiling] SPU execution add completed, input processing took 1.049e-06s, execution took 0.000400461s, output processing took 1.119e-06s, total time 0.000402629s.
[2024-01-03 17:55:25.015] [info] [api.cc:191] HLO profiling: total time 1.5461e-05
[2024-01-03 17:55:25.015] [info] [api.cc:194] - pphlo.add, executed 1 times, duration 1.5461e-05s, send bytes 0
[2024-01-03 17:55:25.015] [info] [api.cc:191] HAL profiling: total time 1.1418e-05
[2024-01-03 17:55:25.015] [info] [api.cc:194] - i_add, executed 1 times, duration 1.1418e-05s, send bytes 0
[2024-01-03 17:55:25.015] [info] [api.cc:191] MPC profiling: total time 6.095e-06
[2024-01-03 17:55:25.015] [info] [api.cc:194] - add_aa, executed 1 times, duration 6.095e-06s, send bytes 0
[2024-01-03 17:55:25.015] [info] [api.cc:204] Link details: total send bytes 0, send actions 0
[2024-01-03 17:55:25,017] [MainProcess] RunR: builtin_fetch_meta at node:0
[2024-01-03 17:55:25,018] [MainProcess] RunR: builtin_fetch_object at node:0
[2024-01-03 17:55:25,020] [MainProcess] RunR: builtin_fetch_object at node:0

### Standalone code to reproduce the issue
import numpy as np
import jax.numpy as jnp

def make_rand():
    np.random.seed()
    return np.random.randint(100, size=(1,))

def add(x, y):
    return jnp.add(x, y)

import json
import os
import io
import argparse

parser = argparse.ArgumentParser(description='SPU node service.')
parser.add_argument("-c", "--config", default="2pc.json", help="the config")
args = parser.parse_args()

if args.config:
    with open(args.config, 'r') as file:
        conf = json.load(file)
    nodes_def = conf["nodes"]
    devices_def = conf["devices"]

import spu.utils.distributed as ppd

ppd.init(nodes_def, devices_def)

x = ppd.device("P1")(make_rand)()

y = ppd.device("P2")(make_rand)()

ans = ppd.device("SPU")(add)(x, y)

print(ppd.get(x))
print(ppd.get(y))
print(ppd.get(ans))

**Comment by tpppppub** [comment]
查看秘密分享分片需要侵入式修改 SPU C++ 代码，自己在需要的位置增加日志，然后源码编译执行

**Comment by daydayuphere** [comment]
那怎么添加呢，不太懂协议。能否给个样例

**Comment by daydayuphere** [comment]
代码的结构有介绍的材料吗，需要先学习下

**Comment by tpppppub** [comment]
可以看看隐语官网的 SPU 文档和 MOOC

**Comment by fionser** [comment]
在什么情况下需要看到具体的收发的分片的值？

**Comment by daydayuphere** [comment]
检测，提供安全证明，代码，日志和抓包一致性的

**Comment by fionser** [comment]
譬如 简单粗暴地抓包 brpc 的流量呢?

**Comment by daydayuphere** [comment]
抓包是其中一种方式，日志也要有

**Comment by anakinxc** [comment]
> 8DEC6DBE38EA876F

把这些 hex 转成 decimal number

一方
lhs = 1846924996263230692
rhs = 8220074556091889857

另一方
lhs = -1846924996263230671
rhs = -8220074556091889809

一方的 lhs + 另一方的 lhs = 21
一方的 rhs + 另一方的 rhs = 48
